src/main/scala/it/pagopa/interop/agreementprocess/error/ErrorHandlers.scala

```diff
@@ -0,0 +1,145 @@
+package it.pagopa.interop.agreementprocess.error
+
+import akka.http.scaladsl.marshalling.ToEntityMarshaller
+import akka.http.scaladsl.model.{StatusCode, StatusCodes}
+import akka.http.scaladsl.server.Directives.complete
+import akka.http.scaladsl.server.StandardRoute
+import com.typesafe.scalalogging.LoggerTakingImplicit
+import it.pagopa.interop.agreementprocess.api.impl.problemOf
+import it.pagopa.interop.agreementprocess.error.AgreementProcessErrors.{
+  AgreementActivationFailed,
+  AgreementAlreadyExists,
+  AgreementNotFound,
+  AgreementNotInExpectedState,
+  DescriptorNotInExpectedState,
+  EServiceNotFound,
+  MissingCertifiedAttributes,
+  MissingDeclaredAttributes,
+  NoNewerDescriptorExists,
+  OperationNotAllowed,
+  PublishedDescriptorNotFound,
+  UnexpectedError
+}
+import it.pagopa.interop.agreementprocess.model.Problem
+import it.pagopa.interop.commons.logging.ContextFieldsToLog
+import it.pagopa.interop.commons.utils.errors.ComponentError
+
+import scala.util.{Failure, Try}
+
+object ErrorHandlers {
```

Comment:
let me know your thoughts

Reply:
_"It's a clean approach and I like it!"_ aka 2023

Reply:
How about making them a bunch of partial functions with another one that is the `orElse` concatenation of all the other ones? That will let us use just an error handler in all the endpoints.

Reply:
not sure if I got your point correctly, but I had to make separate handlers because a specific `ComponentError` does not generate the same status code for each endpoint.
For example `EServiceNotFound` in the agreement creation is a bad request, but during the submission (when we check that the corresponding descriptor is in the correct state) is an internal server error, because it means that we saved an agreement with an eservice that does not exist (or disappeared)